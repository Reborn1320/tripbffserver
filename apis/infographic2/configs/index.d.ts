export namespace InfographicConfig {
  type Background = {
    backgroundColor?: string;
  };
  type WidthHeight = {
    width?: number;
    height?: number;
  };

  type Infographic = {
    width: number;
    height?: number; //maybe just remove it
    blocks: Block[];
  } & Background &
    ContainerBlock;

  type Block = LocationBlock | BasicBlock;
  type BasicBlock =
    | ContainerBlock
    | ImageBlock
    | TextBlock
    | TripNameTextBlock
    | TripInfoTextBlock
    | LocationImageBlock
    | LocationNameTextBlock
    | LocationFeelingTextBlock
    | LocationHightLightsTextBlock;

  type TripNameTextBlock = {
    type: "text";
    text: "{{trip.name}}";
  } & BaseTextBlock;

  type TripInfoTextBlock = {
    type: "text";
    text: "{{trip.info}}";
  } & BaseTextBlock;

  type LocationBlock = {
    type: "location";
    blocks: BasicBlock[];
    positioning?: Positioning;
  } & WidthHeight;

  type LocationNameTextBlock = {
    type: "text";
    text: "{{location.name}}";
  } & BaseTextBlock;

  type LocationFeelingTextBlock = {
    type: "text";
    text: "{{location.feeling}}";
  } & BaseTextBlock;

  type LocationHightLightsTextBlock = {
    type: "text";
    text: "{{location.hight-lights}}";
  } & BaseTextBlock;

  // todo not sure if this is the correct choice
  type ContainerBlock = {
    type: "container";
    blocks: Block[];
    positioning?: Positioning;
  } & Background & WidthHeight;

  type ImageBlock = {
    type: "image";
    url: string;
    // width: number;
    // height: number;
    positioning: Positioning;
  };

  type BaseTextBlock = {
    fontSize?: string;
    fontFamily?: string;
    color?: string;
    fontWeight?: string;
    textAnchor?: string;
    textTransform?: string;
    positioning: Positioning;
  };

  type TextBlock = {
    type: "text";
    text: string;
    width?: number;
  } & BaseTextBlock;

  type LocationImageBlock = {
    type: "location-image";
    width: number;
    height: number;
    clipPath?: string;
    positioning?: Positioning;
  };

  type Positioning = {
    // width?: number;
    // height?: number;
  } & RelativePosition;
  
  // StackPosition | RelativePosition;

  type StackPosition = {
    width?: number;
    height: number;
  };

  type RelativePosition = {
    top?: number;
    left?: number;
    right?: number;
    bottom?: number;
  };

  // type Bounds = {
  //   top?: number;
  //   bottom?: number;
  //   left?: number;
  //   right?: number;
  //   width?: number;
  //   height: number;
  // }
}
