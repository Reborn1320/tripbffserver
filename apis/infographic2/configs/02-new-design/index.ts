import { config02Locations } from "./2-locations";
import { config03Locations } from "./3-locations";
import { configNLocations } from "./n-locations";
import { config01Location } from "./1-location";

export const fullNewDesignConfig = {
  1: config01Location,
  2: config02Locations,
  3: config03Locations,
  n: configNLocations
};