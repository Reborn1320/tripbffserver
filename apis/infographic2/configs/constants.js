var LOCALE_EN = "en";
var LOCALE_VI = "vi";

var FEELING_LABEL_EN = "Feeling";
var FEELING_LABEL_VI = "Cảm thấy";

var DAY_LABEL_EN = "day";
var DAYS_LABEL_EN = "days";
var DAY_LABEL_VI = "ngày";

var LOCATION_LABEL_EN = "location";
var LOCATIONS_LABEL_EN = "locations";
var LOCATION_LABEL_VI = "địa điểm";

module.exports = {
  FEELING_LABEL_EN,
  FEELING_LABEL_VI,

  DAY_LABEL_EN,
  DAYS_LABEL_EN,
  DAY_LABEL_VI,

  LOCATION_LABEL_EN,
  LOCATIONS_LABEL_EN,
  LOCATION_LABEL_VI,

  LOCALE_EN,
  LOCALE_VI
}