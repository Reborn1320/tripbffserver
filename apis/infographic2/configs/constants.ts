export const LOCALE_EN = "en";
export const LOCALE_VI = "vi";

export const FEELING_LABEL_EN = "Feeling";
export const FEELING_LABEL_VI = "Cảm thấy";

export const DAY_LABEL_EN = "day";
export const DAYS_LABEL_EN = "days";
export const DAY_LABEL_VI = "ngày";

export const LOCATION_LABEL_EN = "location";
export const LOCATIONS_LABEL_EN = "locations";
export const LOCATION_LABEL_VI = "địa điểm";
