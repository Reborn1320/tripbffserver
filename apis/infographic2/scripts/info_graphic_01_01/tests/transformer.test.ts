import { InfographicConfig } from "../../../configs";
import { preProcessInfographicConfig } from "../transformer";

describe("node transformer", () => {
  test("simple container node", () => {
    //Arrange
    const config: InfographicConfig.Infographic = {
      width: 1280,
      backgroundColor: "black",
      type: "container",
      blocks: [
        {
          type: "container",
          backgroundColor: "grey",
          height: 100,
          blocks: []
        },
        {
          type: "container",
          backgroundColor: "green",
          height: 1000,
          blocks: []
        }
      ]
    };

    const data = {};

    // Act
    const result = preProcessInfographicConfig(config, data) as InfographicConfig.Infographic;

    // Assert
    expect(result).toBeDefined();
    expect(result.height).toBe(1100);
    expect(result.blocks.length).toBe(2);
    expect(result).toMatchSnapshot();
  });

  test("not supported container node", () => {
    //Arrange
    const config = {
      width: 1280,
      backgroundColor: "black",
      type: "container2",
      blocks: []
    };

    const data = {};

    // Act
    const result = preProcessInfographicConfig(config as InfographicConfig.Infographic, data) as InfographicConfig.Infographic;

    // Assert
    expect(result).toBeDefined();
    expect(result).toBe(config);
  });

  test("simple leaf node", () => {
    //Arrange
    const config: InfographicConfig.Infographic = {
      width: 1280,
      backgroundColor: "black",
      type: "container",
      blocks: [
        {
          type: "container",
          backgroundColor: "grey",
          height: 100,
          blocks: []
        },
        {
          type: "container",
          backgroundColor: "green",
          height: 1000,
          blocks: []
        }
      ]
    };

    const data = {};

    // Act
    const result = preProcessInfographicConfig(config, data) as InfographicConfig.Infographic;

    // Assert
    expect(result).toBeDefined();
    expect(result.height).toBe(1100);
    expect(result.blocks.length).toBe(2);
    expect(result).toMatchSnapshot();
  });
  
});
