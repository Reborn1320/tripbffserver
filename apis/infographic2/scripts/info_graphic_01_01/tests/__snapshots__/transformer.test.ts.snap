// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node transformer simple container node 1`] = `
Object {
  "backgroundColor": "black",
  "blocks": Array [
    Object {
      "backgroundColor": "grey",
      "blocks": Array [],
      "height": 100,
      "type": "container",
    },
    Object {
      "backgroundColor": "green",
      "blocks": Array [],
      "height": 1000,
      "type": "container",
    },
  ],
  "height": 1100,
  "type": "container",
  "width": 1280,
}
`;

exports[`node transformer simple leaf node 1`] = `
Object {
  "backgroundColor": "black",
  "blocks": Array [
    Object {
      "backgroundColor": "grey",
      "blocks": Array [],
      "height": 100,
      "type": "container",
    },
    Object {
      "backgroundColor": "green",
      "blocks": Array [],
      "height": 1000,
      "type": "container",
    },
  ],
  "height": 1100,
  "type": "container",
  "width": 1280,
}
`;
